### PR TITLE
initial work

### PR DESCRIPTION
**Pull request by RDIL**
literally the bare bones of the refactor, obviously not done

### CODE REVIEW

**Comment by N8python** [comment]
Looks pretty good - could you change the result of transpiling `loop` expressions back to `const`, rather than `var`? It is a scope thing.

**Review by N8python — COMMENTED** [review]
Looks pretty good as initial work.

**Review by N8python — CHANGES_REQUESTED** [review]
Looks pretty good as initial work.

File: packages/zcomp/src/compiler/gen.js [hunk 1]
```diff
@@ -388,16 +408,16 @@ function genLoopStatements(loopexpr) {
       curr = anchor;
       return res;
     })();
-    r += `for (const ${iteree} of ${iterable}) {\n`;
+    r += `for (var ${iteree} of ${iterable}) {\n`;
```
Comment (RDIL) [inline packages/zcomp/src/compiler/gen.js:411]:
```suggestion
    r += `for (const ${iteree} of ${iterable}) {\n`;
```

**Pull request merged**